### PR TITLE
style: confirm snake_case for files

### PR DESCRIPTION
**Pull request by matklad**

### CODE REVIEW

**Comment by jorangreef** [comment]
Thanks for adding this, I was about to suggest it for file names too, since it's come up before.

File: docs/TIGER_STYLE.md [hunk 1]
```diff
@@ -116,7 +116,7 @@ Beyond these rules:
 
 * **Get the nouns and verbs just right.** Great names are the essence of great code, they capture what a thing is or does, and provide a crisp, intuitive mental model. They show that you understand the domain. Take time to find the perfect name, to find nouns and verbs that work together, so that the whole is greater than the sum of its parts.
 
-* Use `snake_case` for function and variable names. The underscore is the closest thing we have as programmers to a space, and helps to separate words and encourage descriptive names.
+* Use `snake_case` for function, variable, and file names. The underscore is the closest thing we have as programmers to a space, and helps to separate words and encourage descriptive names.
```
Comment (jorangreef) [inline docs/TIGER_STYLE.md]: Would be great to add:

> Note: We depart from Zig's style here with respect to file names for files containing a type, the rationale being to keep the convention for file names simple and uniform.

**Review by jorangreef — DISMISSED** [review]

File: docs/TIGER_STYLE.md (hunk 1 above)
Comment (matklad) [reply docs/TIGER_STYLE.md]: PTAL

File: docs/TIGER_STYLE.md [hunk 2]
```diff
@@ -116,9 +116,9 @@ Beyond these rules:
 
 * **Get the nouns and verbs just right.** Great names are the essence of great code, they capture what a thing is or does, and provide a crisp, intuitive mental model. They show that you understand the domain. Take time to find the perfect name, to find nouns and verbs that work together, so that the whole is greater than the sum of its parts.
 
-* Use `snake_case` for function and variable names. The underscore is the closest thing we have as programmers to a space, and helps to separate words and encourage descriptive names.
+* Use `snake_case` for function, variable, and file names. The underscore is the closest thing we have as programmers to a space, and helps to separate words and encourage descriptive names. We don't use `CamelCase.zig` for "struct" files to keep the convention simple and consistent with our many "generic struct" files.
```
Comment (jorangreef) [inline docs/TIGER_STYLE.md]:
```suggestion
* Use `snake_case` for function, variable, and file names. The underscore is the closest thing we have as programmers to a space, and helps to separate words and encourage descriptive names. We don't use Zig's `CamelCase.zig` style for "struct" files to keep the convention simple and consistent.
```

**Review by jorangreef — APPROVED** [review]

**Pull request merged**